lib/milestone_date_change.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class MilestoneDateChange
```

Comment:
suggestion: maybe we can move this class into the app/services folder instead so we can reuse and possibly then automate?